transport/server.cc

```diff
@@ -994,19 +996,32 @@ future<std::unique_ptr<cql_server::response>> cql_server::connection::process_st
 }
 
 void cql_server::connection::update_scheduling_group() {
-    if (_client_state.is_control_connection()) {
-        switch_tenant([this] (noncopyable_function<future<> ()> process_loop) -> future<> {
-            auto shg = _server._sl_controller.get_scheduling_group(qos::service_level_controller::driver_service_level_name);
-            _current_scheduling_group = shg;
-            co_return co_await _server._sl_controller.with_service_level(qos::service_level_controller::driver_service_level_name, std::move(process_loop));
-        });
-    } else {
-        switch_tenant([this] (noncopyable_function<future<> ()> process_loop) -> future<> {
-            auto shg = co_await _server._sl_controller.get_user_scheduling_group(_client_state.user());
-            _current_scheduling_group = shg;
-            co_return co_await _server._sl_controller.with_user_service_level(_client_state.user(), std::move(process_loop));
-        });
-    }
+    switch_tenant([this] (noncopyable_function<future<> ()> process_loop) -> future<> {
+        // If the maintenance socket is in use, we cannot rely on *user* service levels.
+        //
+        // The reason is that the service level controller may not have its `auth_integration`
+        // registered. When can a situation like that occur? For instance, when Scylla is
+        // in maintenance mode or when the user attempts to connect to Scylla via the
+        // maintenance socket before the "regular" `auth::service` has been initialized.
+        //
+        // Since *user* service levels rely on the `auth_integration` having been registered,
+        // we can't predict what will happen. In the worst case scenario, Scylla could crash.
+        //
+        // *Regular* service levels are not affected, i.e. we still can reference service levels
+        // by their name. It's only advised against to reference them by a role name in that case.
+        //
+        // If the maintenance socket is in use, the user likely doesn't perform any normal type of
+        // workload, so it should be fine to fall back to the default scheduling group.
```

Comment:
That's the most important thing and it should be at the beginning of the comment. Perhaps the rest of the comment could even be shortened.